components/mdns/mdns.c

```diff
@@ -5200,28 +5200,57 @@ static void _mdns_execute_action(mdns_action_t *action)
         break;
     case ACTION_SERVICE_DEL:
         a = _mdns_server->services;
-        if (action->data.srv_del.service) {
-            if (_mdns_server->services == action->data.srv_del.service) {
-                _mdns_server->services = a->next;
-                _mdns_send_bye(&a, 1, false);
-                _mdns_remove_scheduled_service_packets(a->service);
-                _mdns_free_service(a->service);
-                free(a);
-            } else {
-                while (a->next && a->next != action->data.srv_del.service) {
-                    a = a->next;
+        mdns_srv_item_t *b = a;
+        if (action->data.srv_del.instance) {
```

Comment:
Isn't this a breaking change? The API `mdns_service_remove_for_host()` simply removed the service in the past, whether `instance` was set or not, correct?

Reply:
When `instance` is not set, the service will be removed if matched by `_mdns_service_match()`. And if set, matched by `_mdns_service_match_instance`.

Reply:
That's not what I meant. The changes in this PR make perfect sense.

My only concern is about backward compatibility, so if someone removed services with *some* instance set, it just removed everything. Now it will remove only that single instance, correct?

But this is probably an edge usecase and we can treat this is a bugfix on API level.
Anyway, we have quite a big suite of test cases, so I think we should check if there's a case like that...

Reply:
Yes, this change may result in a different removing result than before...
So should i also add a test case for this?

Reply:
> So should i also add a test case for this?

No need. The test cases I was talking about are not part of this repo.

I think we can just merge it and ask our Q/A team to test it.

Reply:
OK.